com.hazelcast.jet.sql.SqlErrorTest.testTimeout

`frant-hartm:update/avro-1.11.0-5.0.z` (commit d95525c)

Failed on pr-builder: https://jenkins.hazelcast.com/job/Hazelcast-pr-builder/8424/testReport/junit/com.hazelcast.jet.sql/SqlErrorTest/testTimeout/

<details><summary>Stacktrace:</summary>

```
java.lang.AssertionError: Exception is not thrown
	at org.junit.Assert.fail(Assert.java:89)
	at com.hazelcast.jet.sql.SqlErrorAbstractTest.assertSqlException(SqlErrorAbstractTest.java:153)
	at com.hazelcast.jet.sql.SqlErrorAbstractTest.checkTimeout(SqlErrorAbstractTest.java:80)
	at com.hazelcast.jet.sql.SqlErrorTest.testTimeout(SqlErrorTest.java:45)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)

```

</details>

<details><summary>Standard output:</summary>

```
Finished Running Test: testMemberLeave in 1.492 seconds.
Started Running Test: testTimeout
12:59:35,757  INFO |testTimeout| - [MetricsConfigHelper] testTimeout - [LOCAL] [dev] [5.0.1-SNAPSHOT] Overridden metrics configuration with system property 'hazelcast.metrics.collection.frequency'='1' -> 'MetricsConfig.collectionFrequencySeconds'='1'
12:59:35,758  INFO |testTimeout| - [logo] testTimeout - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] 
	+       +  o    o     o     o---o o----o o      o---o     o     o----o o--o--o
	+ +   + +  |    |    / \       /  |      |     /         / \    |         |   
	+ + + + +  o----o   o   o     o   o----o |    o         o   o   o----o    |   
	+ +   + +  |    |  /     \   /    |      |     \       /     \       |    |   
	+       +  o    o o       o o---o o----o o----o o---o o       o o----o    o   
12:59:35,807  INFO |testTimeout| - [system] testTimeout - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
12:59:35,807  INFO |testTimeout| - [system] testTimeout - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Hazelcast Platform 5.0.1-SNAPSHOT (20211201 - dcc48df) starting at [127.0.0.1]:5701
12:59:35,807  INFO |testTimeout| - [system] testTimeout - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Cluster name: dev
12:59:35,819  INFO |testTimeout| - [MetricsConfigHelper] testTimeout - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Collecting debug metrics and sending to diagnostics is enabled
12:59:35,830  WARN |testTimeout| - [CPSubsystem] testTimeout - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] CP Subsystem is not enabled. CP data structures will operate in UNSAFE mode! Please note that UNSAFE mode will not provide strong consistency guarantees.
12:59:35,852  INFO |testTimeout| - [JetServiceBackend] testTimeout - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Setting number of cooperative threads and default parallelism to 2
12:59:35,913  INFO |testTimeout| - [Diagnostics] testTimeout - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Diagnostics disabled. To enable add -Dhazelcast.diagnostics.enabled=true to the JVM arguments.
12:59:35,913  INFO |testTimeout| - [LifecycleService] testTimeout - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] [127.0.0.1]:5701 is STARTING
12:59:35,914  INFO |testTimeout| - [JetExtension] testTimeout - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Jet is enabled
12:59:35,914  INFO |testTimeout| - [ClusterService] testTimeout - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] 

Members {size:1, ver:1} [
	Member [127.0.0.1]:5701 - 14504f45-d48f-41e7-b4c7-3b11149ff96f this
]

12:59:35,914 DEBUG |testTimeout| - [JobCoordinationService] hz.affectionate_babbage.cached.thread-1 - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
12:59:35,914  INFO |testTimeout| - [LifecycleService] testTimeout - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] [127.0.0.1]:5701 is STARTED
12:59:35,917  INFO |testTimeout| - [MetricsConfigHelper] testTimeout - [LOCAL] [dev] [5.0.1-SNAPSHOT] Overridden metrics configuration with system property 'hazelcast.metrics.collection.frequency'='1' -> 'MetricsConfig.collectionFrequencySeconds'='1'
12:59:35,918  INFO |testTimeout| - [logo] testTimeout - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] 
	+       +  o    o     o     o---o o----o o      o---o     o     o----o o--o--o
	+ +   + +  |    |    / \       /  |      |     /         / \    |         |   
	+ + + + +  o----o   o   o     o   o----o |    o         o   o   o----o    |   
	+ +   + +  |    |  /     \   /    |      |     \       /     \       |    |   
	+       +  o    o o       o o---o o----o o----o o---o o       o o----o    o   
12:59:35,918  INFO |testTimeout| - [system] testTimeout - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
12:59:35,918  INFO |testTimeout| - [system] testTimeout - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] Hazelcast Platform 5.0.1-SNAPSHOT (20211201 - dcc48df) starting at [127.0.0.1]:5702
12:59:35,918  INFO |testTimeout| - [system] testTimeout - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] Cluster name: dev
12:59:35,918  INFO |testTimeout| - [HealthMonitor] hz.affectionate_babbage.HealthMonitor - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] processors=16, physical.memory.total=755.6G, physical.memory.free=603.4G, swap.space.total=4.0G, swap.space.free=4.0G, heap.memory.used=513.9M, heap.memory.free=827.6M, heap.memory.total=1.3G, heap.memory.max=1.8G, heap.memory.used/total=38.29%, heap.memory.used/max=28.21%, minor.gc.count=11, minor.gc.time=536ms, major.gc.count=4, major.gc.time=1087ms, load.process=14.29%, load.system=100.00%, load.systemAverage=81.71, thread.count=42, thread.peakCount=147, cluster.timeDiff=0, event.q.size=0, executor.q.async.size=0, executor.q.client.size=0, executor.q.client.query.size=0, executor.q.client.blocking.size=0, executor.q.query.size=0, executor.q.scheduled.size=0, executor.q.io.size=0, executor.q.system.size=0, executor.q.operations.size=0, executor.q.priorityOperation.size=0, operations.completed.count=0, executor.q.mapLoad.size=0, executor.q.mapLoadAllKeys.size=0, executor.q.cluster.size=0, executor.q.response.size=0, operations.running.count=0, operations.pending.invocations.percentage=0.00%, operations.pending.invocations.count=0, proxy.count=1, clientEndpoint.count=0, connection.active.count=0, client.connection.count=0, connection.count=0
12:59:35,928  INFO |testTimeout| - [MetricsConfigHelper] testTimeout - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] Collecting debug metrics and sending to diagnostics is enabled
12:59:35,938  WARN |testTimeout| - [CPSubsystem] testTimeout - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] CP Subsystem is not enabled. CP data structures will operate in UNSAFE mode! Please note that UNSAFE mode will not provide strong consistency guarantees.
12:59:36,008  INFO |testTimeout| - [JetServiceBackend] testTimeout - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] Setting number of cooperative threads and default parallelism to 2
12:59:36,012  INFO |testTimeout| - [Diagnostics] testTimeout - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] Diagnostics disabled. To enable add -Dhazelcast.diagnostics.enabled=true to the JVM arguments.
12:59:36,013  INFO |testTimeout| - [LifecycleService] testTimeout - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] [127.0.0.1]:5702 is STARTING
12:59:36,013  INFO |testTimeout| - [MockServer] testTimeout - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] Created connection to endpoint: [127.0.0.1]:5701, connection: MockConnection{localEndpoint=[127.0.0.1]:5702, remoteEndpoint=[127.0.0.1]:5701, alive=true}
12:59:36,014  INFO |testTimeout| - [MockServer] hz.affectionate_babbage.priority-generic-operation.thread-0 - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Created connection to endpoint: [127.0.0.1]:5702, connection: MockConnection{localEndpoint=[127.0.0.1]:5701, remoteEndpoint=[127.0.0.1]:5702, alive=true}
12:59:36,014 DEBUG |testTimeout| - [JobCoordinationService] hz.affectionate_babbage.cached.thread-1 - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
12:59:36,014  INFO |testTimeout| - [ClusterService] hz.affectionate_babbage.priority-generic-operation.thread-0 - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] 

Members {size:2, ver:2} [
	Member [127.0.0.1]:5701 - 14504f45-d48f-41e7-b4c7-3b11149ff96f this
	Member [127.0.0.1]:5702 - 9936c137-5993-4b78-81fa-9bdc127dcdc1
]

12:59:36,115 DEBUG |testTimeout| - [JobCoordinationService] hz.affectionate_babbage.cached.thread-3 - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
12:59:36,115  INFO |testTimeout| - [JetExtension] hz.nervous_babbage.priority-generic-operation.thread-0 - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] Jet is enabled
12:59:36,116  INFO |testTimeout| - [ClusterService] hz.nervous_babbage.priority-generic-operation.thread-0 - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] 

Members {size:2, ver:2} [
	Member [127.0.0.1]:5701 - 14504f45-d48f-41e7-b4c7-3b11149ff96f
	Member [127.0.0.1]:5702 - 9936c137-5993-4b78-81fa-9bdc127dcdc1 this
]

12:59:36,215 DEBUG |testTimeout| - [JobCoordinationService] hz.affectionate_babbage.cached.thread-3 - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
12:59:36,316 DEBUG |testTimeout| - [JobCoordinationService] hz.affectionate_babbage.cached.thread-3 - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
12:59:36,416 DEBUG |testTimeout| - [JobCoordinationService] hz.affectionate_babbage.cached.thread-2 - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
12:59:36,515  INFO |testTimeout| - [LifecycleService] testTimeout - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] [127.0.0.1]:5702 is STARTED
12:59:36,516 DEBUG |testTimeout| - [JobCoordinationService] hz.affectionate_babbage.cached.thread-2 - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
12:59:36,516  INFO |testTimeout| - [PartitionStateManager] hz.affectionate_babbage.priority-generic-operation.thread-0 - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Initializing cluster partition table arrangement...
12:59:36,518  INFO |testTimeout| - [HealthMonitor] hz.nervous_babbage.HealthMonitor - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] processors=16, physical.memory.total=755.6G, physical.memory.free=603.3G, swap.space.total=4.0G, swap.space.free=4.0G, heap.memory.used=523.0M, heap.memory.free=818.5M, heap.memory.total=1.3G, heap.memory.max=1.8G, heap.memory.used/total=38.93%, heap.memory.used/max=28.69%, minor.gc.count=11, minor.gc.time=536ms, major.gc.count=4, major.gc.time=1087ms, load.process=0.00%, load.system=100.00%, load.systemAverage=81.71, thread.count=66, thread.peakCount=147, cluster.timeDiff=0, event.q.size=0, executor.q.async.size=0, executor.q.client.size=0, executor.q.client.query.size=0, executor.q.client.blocking.size=0, executor.q.query.size=0, executor.q.scheduled.size=0, executor.q.io.size=0, executor.q.system.size=0, executor.q.operations.size=0, executor.q.priorityOperation.size=0, operations.completed.count=12, executor.q.mapLoad.size=0, executor.q.mapLoadAllKeys.size=0, executor.q.cluster.size=0, executor.q.response.size=0, operations.running.count=0, operations.pending.invocations.percentage=0.00%, operations.pending.invocations.count=0, proxy.count=1, clientEndpoint.count=0, connection.active.count=0, client.connection.count=0, connection.count=0
12:59:36,620 DEBUG |testTimeout| - [JobRepository] hz.affectionate_babbage.cached.thread-2 - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Job cleanup took 0ms
12:59:36,723  INFO |testTimeout| - [ClientInvocationService] testTimeout - hz.client_15 [dev] [5.0.1-SNAPSHOT] Running with 2 response threads, dynamic=true
12:59:36,724  INFO |testTimeout| - [LifecycleService] testTimeout - hz.client_15 [dev] [5.0.1-SNAPSHOT] HazelcastClient 5.0.1-SNAPSHOT (20211201 - dcc48df) is STARTING
12:59:36,725  INFO |testTimeout| - [LifecycleService] testTimeout - hz.client_15 [dev] [5.0.1-SNAPSHOT] HazelcastClient 5.0.1-SNAPSHOT (20211201 - dcc48df) is STARTED
12:59:36,726  INFO |testTimeout| - [ClientConnectionManager] testTimeout - hz.client_15 [dev] [5.0.1-SNAPSHOT] Trying to connect to cluster: dev
12:59:36,726  INFO |testTimeout| - [ClientConnectionManager] testTimeout - hz.client_15 [dev] [5.0.1-SNAPSHOT] Trying to connect to [127.0.0.1]:5702
12:59:36,727 DEBUG |testTimeout| - [AuthenticationMessageTask] hz.nervous_babbage.priority-generic-operation.thread-0 - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] Processing authentication with clientUuid 062dedb1-b6a2-428c-a37d-cc653512d45f and clientName hz.client_15
12:59:36,727  INFO |testTimeout| - [AuthenticationMessageTask] hz.nervous_babbage.priority-generic-operation.thread-0 - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] Received auth from MockedNodeConnection{ remoteAddress = [127.0.0.1]:40001, localAddress = [127.0.0.1]:5702, connectionId = 1}, successfully authenticated, clientUuid: 062dedb1-b6a2-428c-a37d-cc653512d45f, client name: hz.client_15, client version: 5.0.1-SNAPSHOT
12:59:36,728  INFO |testTimeout| - [LifecycleService] testTimeout - hz.client_15 [dev] [5.0.1-SNAPSHOT] HazelcastClient 5.0.1-SNAPSHOT (20211201 - dcc48df) is CLIENT_CONNECTED
12:59:36,728  INFO |testTimeout| - [ClientConnectionManager] testTimeout - hz.client_15 [dev] [5.0.1-SNAPSHOT] Authenticated with server [127.0.0.1]:5702:9936c137-5993-4b78-81fa-9bdc127dcdc1, server version: 5.0.1-SNAPSHOT, local address: /127.0.0.1:40001
12:59:36,728  INFO |testTimeout| - [Diagnostics] testTimeout - hz.client_15 [dev] [5.0.1-SNAPSHOT] Diagnostics disabled. To enable add -Dhazelcast.diagnostics.enabled=true to the JVM arguments.
12:59:36,729 TRACE |testTimeout| - [ClientListenerService] hz.client_15.internal-1 - hz.client_15 [dev] [5.0.1-SNAPSHOT] Register attempt of ClusterViewListenerHandler to MockedClientConnection{localAddress=[127.0.0.1]:40001, super=ClientConnection{alive=true, connectionId=1, channel=null, remoteAddress=[127.0.0.1]:5702, lastReadTime=2021-12-01 12:59:36.727, lastWriteTime=2021-12-01 12:59:36.726, closedTime=never, connected server version=5.0.1-SNAPSHOT}}
12:59:36,730  INFO |testTimeout| - [ClientClusterService] hz.client_15.event-109 - hz.client_15 [dev] [5.0.1-SNAPSHOT] 

Members [2] {
	Member [127.0.0.1]:5701 - 14504f45-d48f-41e7-b4c7-3b11149ff96f
	Member [127.0.0.1]:5702 - 9936c137-5993-4b78-81fa-9bdc127dcdc1
}

12:59:36,730 TRACE |testTimeout| - [ClientListenerService] ForkJoinPool.commonPool-worker-11 - hz.client_15 [dev] [5.0.1-SNAPSHOT] Registered ClusterViewListenerHandler to MockedClientConnection{localAddress=[127.0.0.1]:40001, super=ClientConnection{alive=true, connectionId=1, channel=null, remoteAddress=[127.0.0.1]:5702, lastReadTime=2021-12-01 12:59:36.730, lastWriteTime=2021-12-01 12:59:36.729, closedTime=never, connected server version=5.0.1-SNAPSHOT}}
12:59:36,739 DEBUG |testTimeout| - [AuthenticationMessageTask] hz.affectionate_babbage.priority-generic-operation.thread-0 - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Processing authentication with clientUuid 062dedb1-b6a2-428c-a37d-cc653512d45f and clientName hz.client_15
12:59:36,740  INFO |testTimeout| - [AuthenticationMessageTask] hz.affectionate_babbage.priority-generic-operation.thread-0 - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Received auth from MockedNodeConnection{ remoteAddress = [127.0.0.1]:40002, localAddress = [127.0.0.1]:5701, connectionId = 2}, successfully authenticated, clientUuid: 062dedb1-b6a2-428c-a37d-cc653512d45f, client name: hz.client_15, client version: 5.0.1-SNAPSHOT
12:59:36,740  INFO |testTimeout| - [ClientConnectionManager] testTimeout - hz.client_15 [dev] [5.0.1-SNAPSHOT] Authenticated with server [127.0.0.1]:5701:14504f45-d48f-41e7-b4c7-3b11149ff96f, server version: 5.0.1-SNAPSHOT, local address: /127.0.0.1:40002
12:59:36,742 TRACE |testTimeout| - [ClientListenerService] hz.client_15.eventRegistration- - hz.client_15 [dev] [5.0.1-SNAPSHOT] Register attempt of ClientListenerRegistration{codec=com.hazelcast.client.impl.spi.impl.ClientInvocationServiceImpl$1@4a919a37, handler=com.hazelcast.client.impl.spi.impl.ClientInvocationServiceImpl$BackupEventHandler@28d57238} to MockedClientConnection{localAddress=[127.0.0.1]:40001, super=ClientConnection{alive=true, connectionId=1, channel=null, remoteAddress=[127.0.0.1]:5702, lastReadTime=2021-12-01 12:59:36.730, lastWriteTime=2021-12-01 12:59:36.729, closedTime=never, connected server version=5.0.1-SNAPSHOT}}
12:59:36,743 TRACE |testTimeout| - [ClientListenerService] hz.client_15.eventRegistration- - hz.client_15 [dev] [5.0.1-SNAPSHOT] Registered ClientListenerRegistration{codec=com.hazelcast.client.impl.spi.impl.ClientInvocationServiceImpl$1@4a919a37, handler=com.hazelcast.client.impl.spi.impl.ClientInvocationServiceImpl$BackupEventHandler@28d57238} to MockedClientConnection{localAddress=[127.0.0.1]:40001, super=ClientConnection{alive=true, connectionId=1, channel=null, remoteAddress=[127.0.0.1]:5702, lastReadTime=2021-12-01 12:59:36.742, lastWriteTime=2021-12-01 12:59:36.742, closedTime=never, connected server version=5.0.1-SNAPSHOT}}
12:59:36,743 TRACE |testTimeout| - [ClientListenerService] hz.client_15.eventRegistration- - hz.client_15 [dev] [5.0.1-SNAPSHOT] Register attempt of ClientListenerRegistration{codec=com.hazelcast.client.impl.spi.impl.ClientInvocationServiceImpl$1@4a919a37, handler=com.hazelcast.client.impl.spi.impl.ClientInvocationServiceImpl$BackupEventHandler@28d57238} to MockedClientConnection{localAddress=[127.0.0.1]:40002, super=ClientConnection{alive=true, connectionId=2, channel=null, remoteAddress=[127.0.0.1]:5701, lastReadTime=2021-12-01 12:59:36.740, lastWriteTime=2021-12-01 12:59:36.738, closedTime=never, connected server version=5.0.1-SNAPSHOT}}
12:59:36,743 TRACE |testTimeout| - [ClientListenerService] hz.client_15.eventRegistration- - hz.client_15 [dev] [5.0.1-SNAPSHOT] Registered ClientListenerRegistration{codec=com.hazelcast.client.impl.spi.impl.ClientInvocationServiceImpl$1@4a919a37, handler=com.hazelcast.client.impl.spi.impl.ClientInvocationServiceImpl$BackupEventHandler@28d57238} to MockedClientConnection{localAddress=[127.0.0.1]:40002, super=ClientConnection{alive=true, connectionId=2, channel=null, remoteAddress=[127.0.0.1]:5701, lastReadTime=2021-12-01 12:59:36.743, lastWriteTime=2021-12-01 12:59:36.743, closedTime=never, connected server version=5.0.1-SNAPSHOT}}
12:59:36,774 DEBUG |testTimeout| - [CalciteSqlOptimizer] testTimeout - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Before logical opt:
LogicalProject(__key=[$0], this=[$1])
  LogicalTableScan(table=[[hazelcast, public, map[projects=[0, 1]]]])

12:59:36,775 DEBUG |testTimeout| - [CalciteSqlOptimizer] testTimeout - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] After logical opt:
FullScanLogicalRel(table=[[hazelcast, public, map[projects=[0, 1]]]])

12:59:36,775 DEBUG |testTimeout| - [CalciteSqlOptimizer] testTimeout - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] After physical opt:
FullScanPhysicalRel(table=[[hazelcast, public, map[projects=[0, 1]]]])

12:59:36,776 DEBUG |testTimeout| - [LightMasterContext] testTimeout - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Start executing light job 0731-c3ea-7200-0001, execution graph in DOT format:
digraph DAG {
	"IMap[public.map]" [localParallelism=1];
	"Project(IMap[public.map])" [localParallelism=default];
	"ClientSink" [localParallelism=1];
	"IMap[public.map]" -> "Project(IMap[public.map])" [label="isolated", queueSize=1024];
	"Project(IMap[public.map])" -> "ClientSink" [label="distributed to [127.0.0.1]:5701-partitioned", queueSize=1024];
}
HINT: You can use graphviz or http://viz-js.com to visualize the printed graph.
12:59:36,776 DEBUG |testTimeout| - [LightMasterContext] testTimeout - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Building execution plan for 0731-c3ea-7200-0001
12:59:36,777 DEBUG |testTimeout| - [LightMasterContext] testTimeout - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Built execution plans for 0731-c3ea-7200-0001
12:59:36,777 DEBUG |testTimeout| - [InitExecutionOperation] hz.nervous_babbage.generic-operation.thread-1 - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] Initializing execution plan for job 0731-c3ea-7200-0001, execution 0731-c3ea-7200-0001 from [127.0.0.1]:5701
12:59:36,777 DEBUG |testTimeout| - [InitExecutionOperation] testTimeout - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Initializing execution plan for job 0731-c3ea-7200-0001, execution 0731-c3ea-7200-0001 from [127.0.0.1]:5701
12:59:36,778 DEBUG |testTimeout| - [JobExecutionService] hz.nervous_babbage.generic-operation.thread-1 - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] Execution plan for light job ID=0731-c3ea-7200-0001, jobName='0731-c3ea-7200-0001', executionId=0731-c3ea-7200-0001 initialized, will start the execution
12:59:36,778 DEBUG |testTimeout| - [JobExecutionService] testTimeout - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Execution plan for light job ID=0731-c3ea-7200-0001, jobName='0731-c3ea-7200-0001', executionId=0731-c3ea-7200-0001 initialized, will start the execution
12:59:36,866 DEBUG || - [JobExecutionService] ForkJoinPool.commonPool-worker-11 - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] Completed execution of job '0731-c3ea-7200-0001', execution 0731-c3ea-7200-0001
12:59:36,866 DEBUG || - [JobExecutionService] ForkJoinPool.commonPool-worker-11 - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] Execution of job '0731-c3ea-7200-0001', execution 0731-c3ea-7200-0001 completed
12:59:36,878 DEBUG || - [JobExecutionService] ForkJoinPool.commonPool-worker-11 - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Completed execution of job '0731-c3ea-7200-0001', execution 0731-c3ea-7200-0001
12:59:36,878 DEBUG || - [JobExecutionService] ForkJoinPool.commonPool-worker-11 - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Execution of job '0731-c3ea-7200-0001', execution 0731-c3ea-7200-0001 completed
12:59:37,157  INFO |testTimeout| - [LifecycleService] Time-limited test - hz.client_15 [dev] [5.0.1-SNAPSHOT] HazelcastClient 5.0.1-SNAPSHOT (20211201 - dcc48df) is SHUTTING_DOWN
12:59:37,160  INFO |testTimeout| - [MockServer] pool-41-thread-1 - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] Removed connection to endpoint: [127.0.0.1]:40001, connection: MockedNodeConnection{ remoteAddress = [127.0.0.1]:40001, localAddress = [127.0.0.1]:5702, connectionId = 1}
12:59:37,160  INFO |testTimeout| - [ClientConnectionManager] Time-limited test - hz.client_15 [dev] [5.0.1-SNAPSHOT] Removed connection to endpoint: [127.0.0.1]:5702:9936c137-5993-4b78-81fa-9bdc127dcdc1, connection: MockedClientConnection{localAddress=[127.0.0.1]:40001, super=ClientConnection{alive=false, connectionId=1, channel=null, remoteAddress=[127.0.0.1]:5702, lastReadTime=2021-12-01 12:59:36.752, lastWriteTime=2021-12-01 12:59:36.752, closedTime=2021-12-01 12:59:37.160, connected server version=5.0.1-SNAPSHOT}}
12:59:37,161  INFO |testTimeout| - [ClientConnectionManager] Time-limited test - hz.client_15 [dev] [5.0.1-SNAPSHOT] Removed connection to endpoint: [127.0.0.1]:5701:14504f45-d48f-41e7-b4c7-3b11149ff96f, connection: MockedClientConnection{localAddress=[127.0.0.1]:40002, super=ClientConnection{alive=false, connectionId=2, channel=null, remoteAddress=[127.0.0.1]:5701, lastReadTime=2021-12-01 12:59:36.743, lastWriteTime=2021-12-01 12:59:36.743, closedTime=2021-12-01 12:59:37.160, connected server version=5.0.1-SNAPSHOT}}
12:59:37,161  INFO |testTimeout| - [LifecycleService] Time-limited test - hz.client_15 [dev] [5.0.1-SNAPSHOT] HazelcastClient 5.0.1-SNAPSHOT (20211201 - dcc48df) is CLIENT_DISCONNECTED
12:59:37,162  INFO |testTimeout| - [LifecycleService] Time-limited test - hz.client_15 [dev] [5.0.1-SNAPSHOT] HazelcastClient 5.0.1-SNAPSHOT (20211201 - dcc48df) is SHUTDOWN
12:59:37,162  INFO |testTimeout| - [LifecycleService] Time-limited test - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] [127.0.0.1]:5702 is SHUTTING_DOWN
12:59:37,163 DEBUG |testTimeout| - [JobCoordinationService] hz.affectionate_babbage.priority-generic-operation.thread-0 - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Added a shutting-down member: 9936c137-5993-4b78-81fa-9bdc127dcdc1
12:59:37,163  INFO |testTimeout| - [MigrationManager] hz.affectionate_babbage.priority-generic-operation.thread-0 - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Shutdown request of Member [127.0.0.1]:5702 - 9936c137-5993-4b78-81fa-9bdc127dcdc1 is handled
12:59:37,165  INFO |testTimeout| - [MigrationManager] hz.affectionate_babbage.migration - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Repartitioning cluster data. Migration tasks count: 5
12:59:37,166  INFO |testTimeout| - [MockServer] pool-43-thread-1 - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Removed connection to endpoint: [127.0.0.1]:40002, connection: MockedNodeConnection{ remoteAddress = [127.0.0.1]:40002, localAddress = [127.0.0.1]:5701, connectionId = 2}
12:59:37,167  INFO |testTimeout| - [ClientEndpointManager] hz.affectionate_babbage.event-105 - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Destroying ClientEndpoint{connection=MockedNodeConnection{ remoteAddress = [127.0.0.1]:40002, localAddress = [127.0.0.1]:5701, connectionId = 2}, clientUuid=062dedb1-b6a2-428c-a37d-cc653512d45f, clientName=hz.client_15, authenticated=true, clientVersion=5.0.1-SNAPSHOT, creationTime=1638363576739, latest clientAttributes=null, labels=[]}
12:59:37,193  INFO |testTimeout| - [MigrationManager] hz.affectionate_babbage.migration - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] All migration tasks have been completed. (repartitionTime=Wed Dec 01 12:59:37 UTC 2021, plannedMigrations=5, completedMigrations=5, remainingMigrations=0, totalCompletedMigrations=5, elapsedMigrationOperationTime=94ms, totalElapsedMigrationOperationTime=94ms, elapsedDestinationCommitTime=3ms, totalElapsedDestinationCommitTime=3ms, elapsedMigrationTime=106ms, totalElapsedMigrationTime=106ms)
12:59:37,193  INFO |testTimeout| - [Node] Time-limited test - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] Shutting down connection manager...
12:59:37,196  INFO |testTimeout| - [MockServer] Time-limited test - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Removed connection to endpoint: [127.0.0.1]:5702, connection: MockConnection{localEndpoint=[127.0.0.1]:5701, remoteEndpoint=[127.0.0.1]:5702, alive=false}
12:59:37,196  INFO |testTimeout| - [MockServer] Time-limited test - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] Removed connection to endpoint: [127.0.0.1]:5701, connection: MockConnection{localEndpoint=[127.0.0.1]:5702, remoteEndpoint=[127.0.0.1]:5701, alive=false}
12:59:37,196  INFO |testTimeout| - [MembershipManager] Time-limited test - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Removing Member [127.0.0.1]:5702 - 9936c137-5993-4b78-81fa-9bdc127dcdc1
12:59:37,196  INFO |testTimeout| - [ClusterService] Time-limited test - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] 

Members {size:1, ver:3} [
	Member [127.0.0.1]:5701 - 14504f45-d48f-41e7-b4c7-3b11149ff96f this
]

12:59:37,196  INFO |testTimeout| - [Node] Time-limited test - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] Shutting down node engine...
12:59:37,198  INFO |testTimeout| - [TransactionManagerService] hz.affectionate_babbage.cached.thread-7 - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Committing/rolling-back live transactions of [127.0.0.1]:5702, UUID: 9936c137-5993-4b78-81fa-9bdc127dcdc1
12:59:37,198 DEBUG |testTimeout| - [JobCoordinationService] hz.affectionate_babbage.cached.thread-7 - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Removed a shutting-down member: 9936c137-5993-4b78-81fa-9bdc127dcdc1, now shuttingDownMembers=[]
12:59:37,200  INFO |testTimeout| - [MigrationManager] hz.affectionate_babbage.migration - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Partition balance is ok, no need to repartition.
12:59:37,203  INFO |testTimeout| - [NodeExtension] Time-limited test - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] Destroying node NodeExtension.
12:59:37,203  INFO |testTimeout| - [Node] Time-limited test - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] Hazelcast Shutdown is completed in 41 ms.
12:59:37,203  INFO |testTimeout| - [LifecycleService] Time-limited test - [127.0.0.1]:5702 [dev] [5.0.1-SNAPSHOT] [127.0.0.1]:5702 is SHUTDOWN
12:59:37,203  INFO |testTimeout| - [LifecycleService] Time-limited test - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] [127.0.0.1]:5701 is SHUTTING_DOWN
12:59:37,203 DEBUG |testTimeout| - [JobCoordinationService] Time-limited test - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Added a shutting-down member: 14504f45-d48f-41e7-b4c7-3b11149ff96f
12:59:37,203  INFO |testTimeout| - [MigrationManager] hz.affectionate_babbage.cached.thread-7 - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Shutdown request of Member [127.0.0.1]:5701 - 14504f45-d48f-41e7-b4c7-3b11149ff96f this is handled
12:59:37,204  INFO |testTimeout| - [Node] Time-limited test - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Shutting down connection manager...
12:59:37,204  INFO |testTimeout| - [Node] Time-limited test - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Shutting down node engine...
12:59:37,212  INFO |testTimeout| - [NodeExtension] Time-limited test - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Destroying node NodeExtension.
12:59:37,212  INFO |testTimeout| - [Node] Time-limited test - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] Hazelcast Shutdown is completed in 9 ms.
12:59:37,212  INFO |testTimeout| - [LifecycleService] Time-limited test - [127.0.0.1]:5701 [dev] [5.0.1-SNAPSHOT] [127.0.0.1]:5701 is SHUTDOWN
BuildInfo right after testTimeout(com.hazelcast.jet.sql.SqlErrorTest): BuildInfo{version='5.0.1-SNAPSHOT', build='20211201', buildNumber=20211201, revision=dcc48df, enterprise=false, serializationVersion=1}
Hiccups measured while running test 'testTimeout(com.hazelcast.jet.sql.SqlErrorTest):'
12:59:35, accumulated pauses: 651 ms, max pause: 183 ms, pauses over 1000 ms: 0
```

</details>

--- a/hazelcast-sql/src/test/java/com/hazelcast/jet/sql/SqlErrorAbstractTest.java
+++ b/hazelcast-sql/src/test/java/com/hazelcast/jet/sql/SqlErrorAbstractTest.java
@@ -67,17 +67,16 @@ public abstract class SqlErrorAbstractTest extends SqlTestSupport {
         return smallInstanceConfig();
     }
 
-    protected void checkTimeout(boolean useClient, int dataSetSize) {
+    protected void checkTimeout(boolean useClient) {
         // Start two instances and fill them with data
         instance1 = newHazelcastInstance(false);
         instance2 = newHazelcastInstance(true);
         client = newClient();
 
-        createMapping(instance1, MAP_NAME, long.class, long.class);
-        populate(instance1, dataSetSize);
-
         // Execute query on the instance1.
-        HazelcastSqlException error = assertSqlException(useClient ? client : instance1, query().setTimeoutMillis(1L));
+        SqlStatement query = new SqlStatement("select v from table(generate_stream(1))")
+                .setTimeoutMillis(1L);
+        HazelcastSqlException error = assertSqlException(useClient ? client : instance1, query);
         assertTrue(error.getMessage().contains("CANCEL_FORCEFUL"));
     }
 

--- a/hazelcast-sql/src/test/java/com/hazelcast/jet/sql/SqlErrorClientTest.java
+++ b/hazelcast-sql/src/test/java/com/hazelcast/jet/sql/SqlErrorClientTest.java
@@ -88,12 +88,7 @@ public class SqlErrorClientTest extends SqlErrorAbstractTest {
 
     @Test
     public void testTimeout_execute() {
-        checkTimeout(true, DEFAULT_CURSOR_BUFFER_SIZE);
-    }
-
-    @Test
-    public void testTimeout_fetch() {
-        checkTimeout(true, DEFAULT_CURSOR_BUFFER_SIZE * 4);
+        checkTimeout(true);
     }
 
     @Test

--- a/hazelcast-sql/src/test/java/com/hazelcast/jet/sql/SqlErrorTest.java
+++ b/hazelcast-sql/src/test/java/com/hazelcast/jet/sql/SqlErrorTest.java
@@ -29,8 +29,6 @@ import org.junit.Test;
 import org.junit.experimental.categories.Category;
 import org.junit.runner.RunWith;
 
-import static com.hazelcast.jet.sql.SqlTestSupport.awaitSingleRunningJob;
-import static com.hazelcast.sql.SqlStatement.DEFAULT_CURSOR_BUFFER_SIZE;
 import static junit.framework.TestCase.assertEquals;
 
 /**
@@ -42,7 +40,7 @@ public class SqlErrorTest extends SqlErrorAbstractTest {
 
     @Test
     public void testTimeout() {
-        checkTimeout(false, DEFAULT_CURSOR_BUFFER_SIZE);
+        checkTimeout(false);
     }
 
     @Test

--- a/hazelcast/src/main/java/com/hazelcast/sql/SqlStatement.java
+++ b/hazelcast/src/main/java/com/hazelcast/sql/SqlStatement.java
@@ -32,10 +32,16 @@ import java.util.Objects;
  * Changes to properties do not affect the behavior of already running statements.
  */
 public final class SqlStatement {
-    /** Value for the timeout that is not set. */
+    /**
+     * Value for the timeout that is not set. The value of {@link
+     * SqlConfig#setStatementTimeoutMillis} will be used.
+     */
     public static final long TIMEOUT_NOT_SET = -1;
 
-    /** Value for the timeout that is disabled. */
+    /**
+     * Value for the timeout that is disabled, meaning there's no time limit to
+     * run a query.
+     */
     public static final long TIMEOUT_DISABLED = 0;
 
     /** Default timeout. */

